packages/nx/src/config/workspaces.ts

```diff
@@ -916,6 +920,103 @@ export function buildWorkspaceConfigurationFromGlobs(
   };
 }
 
+export function mergeTargetConfigurations(
+  projectConfiguration: ProjectConfiguration,
+  target: string,
+  targetDefaults: TargetDefaults[string]
+): TargetConfiguration {
+  const { targets, root } = projectConfiguration;
+  const targetConfiguration = targets?.[target] ?? {};
```

Comment:
?? {} should never happen, right? That target should always be defined. If that's the case, we should probably throw.